src/include/processor/operator/mask.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include <mutex>
+
+#include "storage/store/node_table.h"
+
+namespace kuzu {
+namespace processor {
+
+// Note: All classes in this file are NOT thread-safe.
+struct MaskUtil {
+    static inline common::offset_t getMorselIdx(common::offset_t offset) {
+        return offset >> common::DEFAULT_VECTOR_CAPACITY_LOG_2;
+    }
+};
+
+struct Mask {
```

Comment:
I think it's better to rename Mask to MaskData or MaskBuffer here, and move `setMask` / `isMasked` out of this struct (to `MaskCollection` as utility functions maybe). This struct will be further simplified to only contain a byte array initialized to 0 used for mask.